src/components/I18nBanner.js

```diff
@@ -0,0 +1,121 @@
+import React, { Component } from 'react';
+import locale from 'browser-locale';
+
+import CaretDownSVG from 'svg/CaretDown';
+
+const languages = [
+  {
+    title: 'English',
+    abbr: 'en',
+  }, {
+    title: 'Español',
+    abbr: 'es',
+  }, {
+    title: 'Tiếng Việt',
+    abbr: 'vn',
+  }, {
+    title: '中文',
+    abbr: 'zh',
+  }, {
+    title: 'Chinese Traditional',
+    abbr: 'zh',
+  }, {
+    title: 'عربى',
+    abbr: 'ar',
+  }, {
+    title: '한국어',
+    abbr: 'ko',
+  }, {
+    title: 'اردو',
+    abbr: 'ur',
+  }, {
+    title: 'မြန်မာ',
+    abbr: 'my',
+  },
+]
+
+class I18nBanner extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      activeLanguage: locale().substring(0,2) || 'en', 
+    };
+  }
+
+  languageClassName = (language) => {
+    const classNameBase = `coa-I18nBanner__language`;
+    return language.abbr === this.state.activeLanguage
+             ? `${classNameBase}--active`
+             : ''
+  }
+
+  otherLanguageClassName = () => {
+    const base = `coa-I18nBanner__other-language-options`;
+    return this.state.isOpen ? `${base}--open` : `${base}--closed`;
+  }
+
+  setLangauage = (e) => {
+    e.stopPropagation();
+    e.nativeEvent.stopImmediatePropagation();
+    const languageAbbr = e.target.parentElement.lang;
```

Comment:
this way of getting the element that's clicked seems highly coupled to the dom structure. maybe try e.currentTarget to get the element which had the handler.